src/shell-integration/bash/ghostty.bash

```diff
@@ -20,14 +20,13 @@
 if [[ "$-" != *i* ]] ; then builtin return; fi
 if [ -z "$GHOSTTY_RESOURCES_DIR" ]; then builtin return; fi
 
-# When automatic shell integration is active, we need to manually
-# load the normal bash startup files based on the injected state.
-if [ -n "$GHOSTTY_BASH_INJECT" ]; then
-  builtin declare ghostty_bash_inject="$GHOSTTY_BASH_INJECT"
-  builtin unset GHOSTTY_BASH_INJECT ENV
-
-  # At this point, we're in POSIX mode and rely on the injected
-  # flags to guide is through the rest of the startup sequence.
+# When automatic shell integration is active, we were started in POSIX
+# mode and need to manually recreate the bash startup sequence.
+if [ -z "${GHOSTTY_BASH_STARTUP:+GHOSTTY_BASH_STARTUP}" ]; then
```

Comment:
If the value _is_ set replace it with the literal string "GHOSTTY_BASH_STARTUP"? I'm not sure what this particular expansion is helping with.

Reply:
> `${parameter:+word}`
> If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted.

This `-z` test will succeed when `$GHOSTTY_BASH_STARTUP` is set to any string (even empty). It will fail (and not run the startup sequence) when `$GHOSTTY_BASH_STARTUP` is missing/null/unset. The literal string `GHOSTTY_BASH_STARTUP` is arbitrary here; it could be anything.

This lets us remove the `1` sentinel value from the environment variable, which was previously needed to pass the `-n` (non-empty) test. We could go back to that approach, too; I don't have strong feelings either way. The minor advantage to what I have here is that `$GHOSTTY_BASH_STARTUP` _only_ contains startup flags.

Reply:
More specifically, isn't `[ -z ${var} ]` equivalent to `[ -z "${var:+var}" ]`. If the value is set replacing it doesn't change the result of the test unless I'm missing something.

Reply:
I'm going to take another pass it this to avoid it being too confusing.

Reply:
I decided to drop the `GHOSTTY_BASH_INJECT` changes from this pull request. It now just focuses on the scope-related change.